### Load Dataframe

In [5]:
setwd("A:/Carlson Minnesota/Summer 2016/MSBA 6120 Introduction to Statistics for Data Scientists/Project/Bike-Sharing-Dataset")
bikeshare <- read.csv("day.csv", header=TRUE)
attach(bikeshare)

The following objects are masked from bikeshare (pos = 3):

    atemp, casual, cnt, dteday, holiday, hum, instant, mnth,
    registered, season, temp, weathersit, weekday, windspeed,
    workingday, yr



In [54]:
str(bikeshare)

'data.frame':	731 obs. of  16 variables:
 $ instant   : int  1 2 3 4 5 6 7 8 9 10 ...
 $ dteday    : Factor w/ 731 levels "2011-01-01","2011-01-02",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ season    : int  1 1 1 1 1 1 1 1 1 1 ...
 $ yr        : int  0 0 0 0 0 0 0 0 0 0 ...
 $ mnth      : int  1 1 1 1 1 1 1 1 1 1 ...
 $ holiday   : int  0 0 0 0 0 0 0 0 0 0 ...
 $ weekday   : int  6 0 1 2 3 4 5 6 0 1 ...
 $ workingday: int  0 0 1 1 1 1 1 0 0 1 ...
 $ weathersit: int  2 2 1 1 1 1 2 2 1 1 ...
 $ temp      : num  0.344 0.363 0.196 0.2 0.227 ...
 $ atemp     : num  0.364 0.354 0.189 0.212 0.229 ...
 $ hum       : num  0.806 0.696 0.437 0.59 0.437 ...
 $ windspeed : num  0.16 0.249 0.248 0.16 0.187 ...
 $ casual    : int  331 131 120 108 82 88 148 68 54 41 ...
 $ registered: int  654 670 1229 1454 1518 1518 1362 891 768 1280 ...
 $ cnt       : int  985 801 1349 1562 1600 1606 1510 959 822 1321 ...


In [55]:
summary(bikeshare)

    instant             dteday        season            yr        
 Min.   :  1.0   2011-01-01:  1   Min.   :1.000   Min.   :0.0000  
 1st Qu.:183.5   2011-01-02:  1   1st Qu.:2.000   1st Qu.:0.0000  
 Median :366.0   2011-01-03:  1   Median :3.000   Median :1.0000  
 Mean   :366.0   2011-01-04:  1   Mean   :2.497   Mean   :0.5007  
 3rd Qu.:548.5   2011-01-05:  1   3rd Qu.:3.000   3rd Qu.:1.0000  
 Max.   :731.0   2011-01-06:  1   Max.   :4.000   Max.   :1.0000  
                 (Other)   :725                                   
      mnth          holiday           weekday        workingday   
 Min.   : 1.00   Min.   :0.00000   Min.   :0.000   Min.   :0.000  
 1st Qu.: 4.00   1st Qu.:0.00000   1st Qu.:1.000   1st Qu.:0.000  
 Median : 7.00   Median :0.00000   Median :3.000   Median :1.000  
 Mean   : 6.52   Mean   :0.02873   Mean   :2.997   Mean   :0.684  
 3rd Qu.:10.00   3rd Qu.:0.00000   3rd Qu.:5.000   3rd Qu.:1.000  
 Max.   :12.00   Max.   :1.00000   Max.   :6.000   Max.   :1.0

### Multiple Regression

In [6]:
set.seed(1)

#Model with all variables
r1 <- lm(cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + workingday + 
           factor(weathersit) + temp + atemp + hum + windspeed)
summary(r1)


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + workingday + factor(weathersit) + temp + 
    atemp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3944.7  -348.2    63.8   457.4  2912.7 

Coefficients: (1 not defined because of singularities)
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1485.84     239.75   6.198 9.77e-10 ***
factor(season)2       884.71     179.49   4.929 1.03e-06 ***
factor(season)3       832.70     213.13   3.907 0.000102 ***
factor(season)4      1575.35     181.00   8.704  < 2e-16 ***
yr                   2019.74      58.22  34.691  < 2e-16 ***
factor(mnth)2         131.03     143.78   0.911 0.362443    
factor(mnth)3         542.83     165.43   3.281 0.001085 ** 
factor(mnth)4         451.17     247.57   1.822 0.068820 .  
factor(mnth)5         735.51     267.63   2.748 0.006145 ** 
factor(mnth)6         515.40     282.41   1.825 0.068423 .  
factor(mnth)

In our model, since workingday predictor is directly linked to the weekday predictor, it has been dropped in R for the 
multiple regression. We deem weekday to be a more valuable indicator as it provides analysis at a more granular level than 
just knowing whether it was a weekday or not and hence proceed with dropping workingday from our model as a predictor. 

Also note that for the factor based predictors (season, mnth, weekday & weathersit), atleast one of the levels has a 
significant p-value at alpha=0.5 and hence there is no need to change the base for any of the predictors.

In [7]:
set.seed(1)

#Model with all variables (dropped workingday at alpha = 0.05)
r2 <- lm(cnt ~ factor(season)+ yr + factor(mnth) + holiday + factor(weekday) +
           factor(weathersit)+ temp + atemp + hum + windspeed)
summary(r2)


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + factor(weathersit) + temp + atemp + hum + 
    windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3944.7  -348.2    63.8   457.4  2912.7 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1485.84     239.75   6.198 9.77e-10 ***
factor(season)2       884.71     179.49   4.929 1.03e-06 ***
factor(season)3       832.70     213.13   3.907 0.000102 ***
factor(season)4      1575.35     181.00   8.704  < 2e-16 ***
yr                   2019.74      58.22  34.691  < 2e-16 ***
factor(mnth)2         131.03     143.78   0.911 0.362443    
factor(mnth)3         542.83     165.43   3.281 0.001085 ** 
factor(mnth)4         451.17     247.57   1.822 0.068820 .  
factor(mnth)5         735.51     267.63   2.748 0.006145 ** 
factor(mnth)6         515.40     282.41   1.825 0.068423 .  
factor(mnth)7          30.80     313.82   0.098 0.921854    
facto

In [8]:
set.seed(1)

#Model with all variables (dropped workingday, atemp at alpha = 0.05)
r3 <- lm(cnt ~ factor(season)+ yr + factor(mnth) + holiday + factor(weekday) +
           factor(weathersit)+ temp + hum + windspeed)
summary(r3)


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + factor(weathersit) + temp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3960.9  -350.9    74.1   456.0  2919.9 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1543.552    235.129   6.565 1.01e-10 ***
factor(season)2       889.302    179.516   4.954 9.12e-07 ***
factor(season)3       832.236    213.204   3.903 0.000104 ***
factor(season)4      1578.947    181.040   8.722  < 2e-16 ***
yr                   2018.063     58.225  34.660  < 2e-16 ***
factor(mnth)2         136.855    143.747   0.952 0.341396    
factor(mnth)3         545.132    165.480   3.294 0.001036 ** 
factor(mnth)4         456.494    247.615   1.844 0.065667 .  
factor(mnth)5         723.520    267.541   2.704 0.007010 ** 
factor(mnth)6         490.552    281.776   1.741 0.082133 .  
factor(mnth)7           8.404    313.395   0.027 0.978613    
factor

season       yr           mnth         holiday      weekday      
season      1.000000000 -0.001844343  0.831440114 -0.010536659 -0.0030798813
yr         -0.001844343  1.000000000 -0.001792434  0.007954311 -0.0054607652
mnth        0.831440114 -0.001792434  1.000000000  0.019190895  0.0095093129
holiday    -0.010536659  0.007954311  0.019190895  1.000000000 -0.1019602689
weekday    -0.003079881 -0.005460765  0.009509313 -0.101960269  1.0000000000
weathersit  0.019211028 -0.048726541  0.043528098 -0.034626841  0.0310874694
temp        0.334314856  0.047603572  0.220205335 -0.028555535 -0.0001699624
hum         0.205444765 -0.110651045  0.222203691 -0.015937479 -0.0522321004
windspeed  -0.229046337 -0.011817060 -0.207501752  0.006291507  0.0142821241
           weathersit  temp          hum         windspeed   
season      0.01921103  0.3343148564  0.20544476 -0.229046337
yr         -0.04872654  0.0476035719 -0.11065104 -0.011817060
mnth        0.04352810  0.2202053352  0.22220369 -0.207501752
holiday    -0.03462684 -0.0285555350 -0.01593748  0.006291507
weekday     0.03108747 -0.0001699624 -0.05223210  0.014282124
weathersit  1.00000000 -0.1206022365  0.59104460  0.039511059
temp       -0.12060224  1.0000000000  0.12696294 -0.157944120
hum         0.59104460  0.1269629390  1.00000000 -0.248489099
windspeed   0.03951106 -0.1579441204 -0.24848910  1.000000000

Since all predictors are now significant we want to check correlation such that our beta's should be interpretable

In [12]:
cor(bikeshare[,c("season","yr","mnth","holiday","weekday","weathersit","temp","hum","windspeed")])

season       yr           mnth         holiday      weekday      
season      1.000000000 -0.001844343  0.831440114 -0.010536659 -0.0030798813
yr         -0.001844343  1.000000000 -0.001792434  0.007954311 -0.0054607652
mnth        0.831440114 -0.001792434  1.000000000  0.019190895  0.0095093129
holiday    -0.010536659  0.007954311  0.019190895  1.000000000 -0.1019602689
weekday    -0.003079881 -0.005460765  0.009509313 -0.101960269  1.0000000000
weathersit  0.019211028 -0.048726541  0.043528098 -0.034626841  0.0310874694
temp        0.334314856  0.047603572  0.220205335 -0.028555535 -0.0001699624
hum         0.205444765 -0.110651045  0.222203691 -0.015937479 -0.0522321004
windspeed  -0.229046337 -0.011817060 -0.207501752  0.006291507  0.0142821241
           weathersit  temp          hum         windspeed   
season      0.01921103  0.3343148564  0.20544476 -0.229046337
yr         -0.04872654  0.0476035719 -0.11065104 -0.011817060
mnth        0.04352810  0.2202053352  0.22220369 -0.207501752
holiday    -0.03462684 -0.0285555350 -0.01593748  0.006291507
weekday     0.03108747 -0.0001699624 -0.05223210  0.014282124
weathersit  1.00000000 -0.1206022365  0.59104460  0.039511059
temp       -0.12060224  1.0000000000  0.12696294 -0.157944120
hum         0.59104460  0.1269629390  1.00000000 -0.248489099
windspeed   0.03951106 -0.1579441204 -0.24848910  1.000000000

>0.7 (season,holiday)

Next possible drop of highest p-value is that of holiday (0.000845) i.e.choosing between lowest p-value of season (0.000104) 
and that of holiday.Even though significant at alpha=0.05, we check our model by dropping it from our model to make 
beta's interpretable by removing collinearity. 

In [9]:
set.seed(1)

#Model with all variables (dropped workingday, atemp, holiday at alpha = 0.05)
r4 <- lm(cnt ~ factor(season)+ yr + factor(mnth) + factor(weekday) +
           factor(weathersit)+ temp + hum + windspeed)
summary(r4)


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + factor(weekday) + 
    factor(weathersit) + temp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3971.0  -352.1    77.5   459.5  2911.7 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1538.52     236.83   6.496 1.56e-10 ***
factor(season)2       896.12     180.80   4.956 9.01e-07 ***
factor(season)3       849.74     214.68   3.958 8.32e-05 ***
factor(season)4      1630.17     181.70   8.972  < 2e-16 ***
yr                   2019.08      58.65  34.428  < 2e-16 ***
factor(mnth)2         148.98     144.74   1.029 0.303712    
factor(mnth)3         581.86     166.31   3.499 0.000497 ***
factor(mnth)4         476.58     249.34   1.911 0.056359 .  
factor(mnth)5         752.54     269.34   2.794 0.005347 ** 
factor(mnth)6         544.40     283.36   1.921 0.055101 .  
factor(mnth)7          41.65     315.51   0.132 0.895018    
factor(mnth)8         453.31

In [10]:
set.seed(1)

#Model with all variables (dropped workingday, atemp, holiday at alpha = 0.05)
r5 <- lm(cnt ~ factor(mnth)+ yr + factor(weekday) +
           factor(weathersit)+ temp + hum + windspeed)
summary(r5)

The following objects are masked from bikeshare (pos = 3):

    atemp, casual, cnt, dteday, holiday, hum, instant, mnth,
    registered, season, temp, weathersit, weekday, windspeed,
    workingday, yr

The following objects are masked from bikeshare (pos = 4):

    atemp, casual, cnt, dteday, holiday, hum, instant, mnth,
    registered, season, temp, weathersit, weekday, windspeed,
    workingday, yr




Call:
lm(formula = cnt ~ factor(mnth) + yr + factor(weekday) + factor(weathersit) + 
    temp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3954.1  -394.9    68.6   495.7  2611.9 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1557.29     251.09   6.202 9.48e-10 ***
factor(mnth)2         145.70     153.55   0.949  0.34303    
factor(mnth)3         885.57     162.22   5.459 6.63e-08 ***
factor(mnth)4        1352.89     180.18   7.509 1.81e-13 ***
factor(mnth)5        1587.40     210.83   7.529 1.56e-13 ***
factor(mnth)6        1360.24     243.85   5.578 3.46e-08 ***
factor(mnth)7         809.32     267.56   3.025  0.00258 ** 
factor(mnth)8        1226.47     249.23   4.921 1.07e-06 ***
factor(mnth)9        1978.95     219.09   9.032  < 2e-16 ***
factor(mnth)10       2076.94     181.85  11.421  < 2e-16 ***
factor(mnth)11       1438.23     159.49   9.018  < 2e-16 ***
factor(mnth)12        921.44     153.14   6.0

In [11]:
set.seed(1)

#Final Model with 8 variables
rfinal <- lm(cnt~factor(season)+yr+holiday+factor(weathersit)+ workingday+temp+ windspeed + hum)
summary(rfinal)


Call:
lm(formula = cnt ~ factor(season) + yr + holiday + factor(weathersit) + 
    workingday + temp + windspeed + hum)

Residuals:
    Min      1Q  Median      3Q     Max 
-3677.3  -378.9    67.4   475.7  3347.1 

Coefficients:
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1586.95     226.75   6.999 5.93e-12 ***
factor(season)2      1137.94     113.55  10.021  < 2e-16 ***
factor(season)3       844.10     149.97   5.628 2.61e-08 ***
factor(season)4      1545.59      96.67  15.988  < 2e-16 ***
yr                   2013.71      61.75  32.610  < 2e-16 ***
holiday              -624.01     188.44  -3.311 0.000974 ***
factor(weathersit)2  -426.72      81.24  -5.253 1.98e-07 ***
factor(weathersit)3 -1916.95     207.31  -9.247  < 2e-16 ***
workingday            118.19      67.92   1.740 0.082232 .  
temp                 5108.07     306.98  16.640  < 2e-16 ***
windspeed           -2795.28     428.06  -6.530 1.24e-10 ***
hum                 -1325.64     295.0

### Stepwise Regression

In [21]:
set.seed(1)

s1 <- step(lm(cnt~factor(season)+yr+factor(mnth)+holiday+factor(weekday)+workingday+factor(weathersit)+temp+atemp+hum+windspeed),
           direction="both")
summary(s1)

Start:  AIC=9744
cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + 
    workingday + factor(weathersit) + temp + atemp + hum + windspeed


Step:  AIC=9744
cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + 
    factor(weathersit) + temp + atemp + hum + windspeed

                     Df Sum of Sq        RSS     AIC
- atemp               1    883090  416284778  9743.6
<none>                             415401688  9744.0
- temp                1   2467375  417869064  9746.3
- holiday             1   6325545  421727233  9753.0
- factor(weekday)     6  15346257  430747945  9758.5
- hum                 1  16312287  431713975  9770.2
- windspeed           1  27446438  442848126  9788.8
- factor(mnth)       11  57051189  472452877  9816.1
- factor(season)      3  51711125  467112813  9823.8
- factor(weathersit)  2  62999538  478401226  9843.2
- yr                  1 712147968 1127549656 10471.9

Step:  AIC=9743.55
cnt ~ factor(season) + yr + factor(mnth


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + factor(weathersit) + temp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3960.9  -350.9    74.1   456.0  2919.9 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1543.552    235.129   6.565 1.01e-10 ***
factor(season)2       889.302    179.516   4.954 9.12e-07 ***
factor(season)3       832.236    213.204   3.903 0.000104 ***
factor(season)4      1578.947    181.040   8.722  < 2e-16 ***
yr                   2018.063     58.225  34.660  < 2e-16 ***
factor(mnth)2         136.855    143.747   0.952 0.341396    
factor(mnth)3         545.132    165.480   3.294 0.001036 ** 
factor(mnth)4         456.494    247.615   1.844 0.065667 .  
factor(mnth)5         723.520    267.541   2.704 0.007010 ** 
factor(mnth)6         490.552    281.776   1.741 0.082133 .  
factor(mnth)7           8.404    313.395   0.027 0.978613    
factor

In [22]:
set.seed(1)

s2 <- step(lm(cnt~factor(season)+yr+factor(mnth)+holiday+factor(weekday)+workingday+factor(weathersit)+temp+atemp+hum+windspeed),
           direction="forward")
summary(s2)

Start:  AIC=9744
cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + 
    workingday + factor(weathersit) + temp + atemp + hum + windspeed




Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + workingday + factor(weathersit) + temp + 
    atemp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3944.7  -348.2    63.8   457.4  2912.7 

Coefficients: (1 not defined because of singularities)
                    Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1485.84     239.75   6.198 9.77e-10 ***
factor(season)2       884.71     179.49   4.929 1.03e-06 ***
factor(season)3       832.70     213.13   3.907 0.000102 ***
factor(season)4      1575.35     181.00   8.704  < 2e-16 ***
yr                   2019.74      58.22  34.691  < 2e-16 ***
factor(mnth)2         131.03     143.78   0.911 0.362443    
factor(mnth)3         542.83     165.43   3.281 0.001085 ** 
factor(mnth)4         451.17     247.57   1.822 0.068820 .  
factor(mnth)5         735.51     267.63   2.748 0.006145 ** 
factor(mnth)6         515.40     282.41   1.825 0.068423 .  
factor(mnth)

In [23]:
set.seed(1)

s3 <- step(lm(cnt~factor(season)+yr+factor(mnth)+holiday+factor(weekday)+workingday+factor(weathersit)+temp+atemp+hum+windspeed),
           direction="backward")
summary(s3)

Start:  AIC=9744
cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + 
    workingday + factor(weathersit) + temp + atemp + hum + windspeed


Step:  AIC=9744
cnt ~ factor(season) + yr + factor(mnth) + holiday + factor(weekday) + 
    factor(weathersit) + temp + atemp + hum + windspeed

                     Df Sum of Sq        RSS     AIC
- atemp               1    883090  416284778  9743.6
<none>                             415401688  9744.0
- temp                1   2467375  417869064  9746.3
- holiday             1   6325545  421727233  9753.0
- factor(weekday)     6  15346257  430747945  9758.5
- hum                 1  16312287  431713975  9770.2
- windspeed           1  27446438  442848126  9788.8
- factor(mnth)       11  57051189  472452877  9816.1
- factor(season)      3  51711125  467112813  9823.8
- factor(weathersit)  2  62999538  478401226  9843.2
- yr                  1 712147968 1127549656 10471.9

Step:  AIC=9743.55
cnt ~ factor(season) + yr + factor(mnth


Call:
lm(formula = cnt ~ factor(season) + yr + factor(mnth) + holiday + 
    factor(weekday) + factor(weathersit) + temp + hum + windspeed)

Residuals:
    Min      1Q  Median      3Q     Max 
-3960.9  -350.9    74.1   456.0  2919.9 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)          1543.552    235.129   6.565 1.01e-10 ***
factor(season)2       889.302    179.516   4.954 9.12e-07 ***
factor(season)3       832.236    213.204   3.903 0.000104 ***
factor(season)4      1578.947    181.040   8.722  < 2e-16 ***
yr                   2018.063     58.225  34.660  < 2e-16 ***
factor(mnth)2         136.855    143.747   0.952 0.341396    
factor(mnth)3         545.132    165.480   3.294 0.001036 ** 
factor(mnth)4         456.494    247.615   1.844 0.065667 .  
factor(mnth)5         723.520    267.541   2.704 0.007010 ** 
factor(mnth)6         490.552    281.776   1.741 0.082133 .  
factor(mnth)7           8.404    313.395   0.027 0.978613    
factor

# Future Work 

### Ridge Regression

In [17]:
library(MASS)

In [57]:
set.seed(1)

lm.ridge(cnt~factor(season)+yr+factor(mnth)+holiday+factor(weekday)+workingday+factor(weathersit)+temp+atemp+hum
                  +windspeed,bikeshare)

                        factor(season)2     factor(season)3     factor(season)4 
       1.506729e+03        8.832914e+02        8.257037e+02        1.569005e+03 
                 yr       factor(mnth)2       factor(mnth)3       factor(mnth)4 
       2.020289e+03        1.399872e+02        5.477330e+02        4.498089e+02 
      factor(mnth)5       factor(mnth)6       factor(mnth)7       factor(mnth)8 
       7.463275e+02        5.067703e+02        3.779684e+01        4.565941e+02 
      factor(mnth)9      factor(mnth)10      factor(mnth)11      factor(mnth)12 
       1.007029e+03        5.283004e+02       -1.029279e+02       -8.855847e+01 
            holiday    factor(weekday)1    factor(weekday)2    factor(weekday)3 
       7.267451e+15       -7.267451e+15       -7.267451e+15       -7.267451e+15 
   factor(weekday)4    factor(weekday)5    factor(weekday)6          workingday 
      -7.267451e+15       -7.267451e+15        4.375683e+02        7.267451e+15 
factor(weathersit)2 factor(w

In [52]:
library(glmnet)

### Lasso Regression

In [51]:
set.seed(1)

l_data <- as.matrix(subset(bikeshare,select=-c(cnt,dteday)))
l_target <- as.matrix(subset(bikeshare,select=c(cnt)))

l1 <- glmnet(l_data, l_target, family='gaussian', alpha=1, standardize=TRUE)
l1
summary(l1)


Call:  glmnet(x = l_data, y = l_target, family = "gaussian", alpha = 1,      standardize = TRUE) 

      Df   %Dev  Lambda
 [1,]  0 0.0000 1830.00
 [2,]  1 0.1518 1668.00
 [3,]  1 0.2778 1520.00
 [4,]  1 0.3824 1385.00
 [5,]  1 0.4693 1262.00
 [6,]  1 0.5414 1150.00
 [7,]  1 0.6013 1047.00
 [8,]  2 0.6516  954.40
 [9,]  2 0.7108  869.60
[10,]  2 0.7599  792.30
[11,]  2 0.8006  722.00
[12,]  2 0.8345  657.80
[13,]  2 0.8626  599.40
[14,]  2 0.8859  546.10
[15,]  2 0.9053  497.60
[16,]  2 0.9214  453.40
[17,]  2 0.9347  413.10
[18,]  2 0.9458  376.40
[19,]  2 0.9550  343.00
[20,]  2 0.9626  312.50
[21,]  2 0.9690  284.80
[22,]  2 0.9743  259.50
[23,]  2 0.9786  236.40
[24,]  2 0.9823  215.40
[25,]  2 0.9853  196.30
[26,]  2 0.9878  178.80
[27,]  2 0.9898  162.90
[28,]  2 0.9916  148.50
[29,]  2 0.9930  135.30
[30,]  2 0.9942  123.30
[31,]  2 0.9952  112.30
[32,]  2 0.9960  102.30
[33,]  2 0.9967   93.24
[34,]  2 0.9972   84.96
[35,]  2 0.9977   77.41
[36,]  2 0.9981   70.54
[37,]  2 0.9

          Length Class     Mode   
a0         40    -none-    numeric
beta      560    dgCMatrix S4     
df         40    -none-    numeric
dim         2    -none-    numeric
lambda     40    -none-    numeric
dev.ratio  40    -none-    numeric
nulldev     1    -none-    numeric
npasses     1    -none-    numeric
jerr        1    -none-    numeric
offset      1    -none-    logical
call        6    -none-    call   
nobs        1    -none-    numeric

### Elastic Net Regression

In [53]:
library(elasticnet)

In [56]:
set.seed(1)

en_data <- as.matrix(subset(bikeshare,select=-c(cnt,dteday)))
en_target <- as.matrix(subset(bikeshare,select=c(cnt)))

en1 <- enet(en_data,en_target,lambda=0.5)
en1
summary(en1)


Call:
enet(x = en_data, y = en_target, lambda = 0.5)
Sequence of  moves:
     registered casual atemp temp instant yr weathersit season windspeed hum
Var          14     13    10    9       1  3          8      2        12  11
Step          1      2     3    4       5  6          7      8         9  10
     workingday weekday holiday mnth   
Var           7       6       5    4 15
Step         11      12      13   14 15

          Length Class  Mode     
call        4    -none- call     
actions    15    -none- list     
allset     14    -none- numeric  
beta.pure 210    -none- numeric  
vn         14    -none- character
mu          1    -none- numeric  
normx      14    -none- numeric  
meanx      14    -none- numeric  
lambda      1    -none- numeric  
L1norm     15    -none- numeric  
penalty    15    -none- numeric  
df         15    -none- numeric  
Cp         15    -none- numeric  
sigma2      1    -none- numeric  